### PR TITLE
feat: Support karpenter.sh/instance-network-bandwidth

### PR DESCRIPTION
**Pull request by ellistarn**
Fixes # 

**Description**

Introduces a well known label for instance-bandwidth. I chose our units to be megabits instead of megabytes (similar to instance-memory), as Kubernetes only supports integer comparison for node selector operators. 

```
affinity:
   nodeAffinity:
     requiredDuringSchedulingIgnoredDuringExecution:
       nodeSelectorTerms:
       - key: karpenter.sh/instance-network-bandwidth
          operator: Gt
          values: [ 10000 ] // Node must have > 10gigabit connection 
```

**How was this change tested?**

* `make e2etests`
```
=== RUN   TestIntegration
Running Suite: Integration - /Users/etarn/workspaces/go/src/github.com/aws/karpenter/test/suites/integration
============================================================================================================
Random Seed: 1677009826

Will run 1 of 68 specs
------------------------------
[BeforeSuite]
/Users/etarn/workspaces/go/src/github.com/aws/karpenter/test/suites/integration/suite_test.go:30
    logger.go:130: 2023-02-21T12:03:48.473-0800	DEBUG	injection/injection.go:102	waiting for configmaps
[BeforeSuite] PASSED [2.172 seconds]
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
Scheduling should support well known labels
/Users/etarn/workspaces/go/src/github.com/aws/karpenter/test/suites/integration/scheduling_test.go:56
------- START BEFORE -------
    logger.go:130: 2023-02-21T12:03:49.092-0800	DEBUG	injection/injection.go:102	waiting for configmaps
node ip-192-168-18-81.us-west-2.compute.internal ready=True schedulable=true initialized= pods=3 taints=[{CriticalAddonsOnly  NoSchedule <nil>}]
node ip-192-168-35-250.us-west-2.compute.internal ready=True schedulable=true initialized= pods=1 taints=[{CriticalAddonsOnly  NoSchedule <nil>}]
pods karpenter/karpenter-dc44459fc-5rdwz provisionable=false phase=Running nodename=ip-192-168-18-81.us-west-2.compute.internal [controller restarts=0]
pods karpenter/karpenter-dc44459fc-8hvfc provisionable=false phase=Running nodename=ip-192-168-35-250.us-west-2.compute.internal [controller restarts=0]
pods kube-system/aws-node-ct8x2 provisionable=false phase=Running nodename=ip-192-168-18-81.us-west-2.compute.internal [aws-node restarts=0]
pods kube-system/aws-node-p9zxx provisionable=false phase=Running nodename=ip-192-168-35-250.us-west-2.compute.internal [aws-node restarts=0]
pods kube-system/coredns-57ff979f67-hdvsz provisionable=false phase=Running nodename=ip-192-168-18-81.us-west-2.compute.internal [coredns restarts=0]
pods kube-system/coredns-57ff979f67-t8b67 provisionable=false phase=Running nodename=ip-192-168-18-81.us-west-2.compute.internal [coredns restarts=0]
pods kube-system/kube-proxy-b6bq9 provisionable=false phase=Running nodename=ip-192-168-18-81.us-west-2.compute.internal [kube-proxy restarts=0]
pods kube-system/kube-proxy-c4chc provisionable=false phase=Running nodename=ip-192-168-35-250.us-west-2.compute.internal [kube-proxy restarts=0]
------- END BEFORE -------
[CREATED 2023-02-21T12:03:49-08:00] pods default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh provisionable=false phase=Pending nodename= []
[UPDATED 2023-02-21T12:03:49-08:00] pods default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh provisionable=true phase=Pending nodename= []
[CREATED 2023-02-21T12:03:54-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready= schedulable=true initialized= pods=0 taints=[{node.kubernetes.io/not-ready  NoSchedule <nil>}]
[UPDATED 2023-02-21T12:03:54-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready= schedulable=true initialized= pods=0 taints=[]
[UPDATED 2023-02-21T12:04:39-08:00] pods default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh provisionable=false phase=Pending nodename=ip-192-168-81-105.us-west-2.compute.internal []
[UPDATED 2023-02-21T12:04:39-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=False schedulable=true initialized= pods=1 taints=[]
[UPDATED 2023-02-21T12:04:39-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=False schedulable=true initialized= pods=1 taints=[{node.kubernetes.io/not-ready  NoSchedule 2023-02-21 12:04:39 -0800 PST}]
[UPDATED 2023-02-21T12:04:40-08:00] pods default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh provisionable=false phase=Pending nodename=ip-192-168-81-105.us-west-2.compute.internal [racerorange-6-9cocwfuaot restarts=0]
[UPDATED 2023-02-21T12:04:40-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=False schedulable=true initialized= pods=1 taints=[{node.kubernetes.io/not-ready  NoSchedule 2023-02-21 12:04:39 -0800 PST} {node.kubernetes.io/not-ready  NoExecute 2023-02-21 12:04:40 -0800 PST}]
[UPDATED 2023-02-21T12:04:57-08:00] pods default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh provisionable=false phase=Running nodename=ip-192-168-81-105.us-west-2.compute.internal [racerorange-6-9cocwfuaot restarts=0]
------- START AWS CLEANUP -------
------- START CLEANUP -------
[UPDATED 2023-02-21T12:04:58-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=False schedulable=false initialized= pods=0 taints=[{node.kubernetes.io/not-ready  NoSchedule 2023-02-21 12:04:39 -0800 PST} {node.kubernetes.io/not-ready  NoExecute 2023-02-21 12:04:40 -0800 PST}]
[UPDATED 2023-02-21T12:04:58-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=False schedulable=false initialized= pods=0 taints=[{node.kubernetes.io/not-ready  NoSchedule 2023-02-21 12:04:39 -0800 PST} {node.kubernetes.io/not-ready  NoExecute 2023-02-21 12:04:40 -0800 PST} {node.kubernetes.io/unschedulable  NoSchedule 2023-02-21 12:04:58 -0800 PST}]
[UPDATED 2023-02-21T12:05:00-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=True schedulable=false initialized= pods=0 taints=[{node.kubernetes.io/not-ready  NoSchedule 2023-02-21 12:04:39 -0800 PST} {node.kubernetes.io/not-ready  NoExecute 2023-02-21 12:04:40 -0800 PST} {node.kubernetes.io/unschedulable  NoSchedule 2023-02-21 12:04:58 -0800 PST}]
[UPDATED 2023-02-21T12:05:00-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=True schedulable=false initialized= pods=0 taints=[{node.kubernetes.io/not-ready  NoExecute 2023-02-21 12:04:40 -0800 PST} {node.kubernetes.io/unschedulable  NoSchedule 2023-02-21 12:04:58 -0800 PST}]
[DELETED 2023-02-21T12:05:00-08:00] pods default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh provisionable=false phase=Running nodename=ip-192-168-81-105.us-west-2.compute.internal [racerorange-6-9cocwfuaot restarts=0]
[UPDATED 2023-02-21T12:05:05-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=True schedulable=false initialized= pods=0 taints=[{node.kubernetes.io/unschedulable  NoSchedule 2023-02-21 12:04:58 -0800 PST}]
[DELETED 2023-02-21T12:05:05-08:00] node ip-192-168-81-105.us-west-2.compute.internal ready=True schedulable=false initialized= pods=0 taints=[{node.kubernetes.io/unschedulable  NoSchedule 2023-02-21 12:04:58 -0800 PST}]
------- END CLEANUP -------
------- END AWS CLEANUP -------
------- START FORCE CLEANUP -------
------- END FORCE CLEANUP -------
------- START AFTER -------
------- pod/default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh EVENTS -------
time=2023-02-21T12:03:50-08:00 type=Warning reason=FailedScheduling from=default-scheduler message=0/2 nodes are available: 2 node(s) had taint {CriticalAddonsOnly: }, that the pod didn't tolerate.
time=2023-02-21T12:03:54-08:00 type=Normal reason=Nominated from=karpenter message=Pod should schedule on ip-192-168-81-105.us-west-2.compute.internal
time=2023-02-21T12:04:39-08:00 type=Normal reason=Scheduled from=default-scheduler message=Successfully assigned default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh to ip-192-168-81-105.us-west-2.compute.internal
time=2023-02-21T12:04:40-08:00 type=Warning reason=NetworkNotReady from=kubelet message=network is not ready: container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized
time=2023-02-21T12:04:40-08:00 type=Warning reason=FailedMount from=kubelet message=MountVolume.SetUp failed for volume "kube-api-access-4bstn" : object "default"/"kube-root-ca.crt" not registered
time=2023-02-21T12:04:56-08:00 type=Normal reason=Pulling from=kubelet message=Pulling image "public.ecr.aws/eks-distro/kubernetes/pause:3.2"
time=2023-02-21T12:04:57-08:00 type=Normal reason=Pulled from=kubelet message=Successfully pulled image "public.ecr.aws/eks-distro/kubernetes/pause:3.2" in 533.633313ms
time=2023-02-21T12:04:57-08:00 type=Normal reason=Created from=kubelet message=Created container racerorange-6-9cocwfuaot
time=2023-02-21T12:04:57-08:00 type=Normal reason=Started from=kubelet message=Started container racerorange-6-9cocwfuaot
time=2023-02-21T12:04:59-08:00 type=Normal reason=Killing from=kubelet message=Stopping container racerorange-6-9cocwfuaot
time=2023-02-21T12:05:00-08:00 type=Normal reason=TaintManagerEviction from=taint-controller message=Cancelling deletion of Pod default/slothglow-4-k7i4ymr18q-5bcdd99c64-gqsmh
------- START CONTROLLER LOGS -------
------- pod/karpenter-dc44459fc-5rdwz -------
    logger.go:130: 2023-02-21T12:05:06.883-0800	INFO	common/expectations.go:298
------- pod/karpenter-dc44459fc-8hvfc -------
    logger.go:130: 2023-02-21T12:05:06.924-0800	INFO	common/expectations.go:298	2023-02-21T20:03:52.297Z	INFO	controller.provisioner	found provisionable pod(s)	{"commit": "dedf6c1", "pods": 1}
        2023-02-21T20:03:52.297Z	INFO	controller.provisioner	computed new node(s) to fit pod(s)	{"commit": "dedf6c1", "nodes": 1, "pods": 1}
        2023-02-21T20:03:52.297Z	INFO	controller.provisioner	launching machine with 1 pods requesting {"cpu":"125m","pods":"3"} from types g4dn.8xlarge	{"commit": "dedf6c1", "provisioner": "biterfossil-2-xzruescu8h"}
        2023-02-21T20:03:52.515Z	DEBUG	controller.provisioner.cloudprovider	created launch template	{"commit": "dedf6c1", "provisioner": "biterfossil-2-xzruescu8h", "launch-template-name": "Karpenter-dev-6910532314892901476", "launch-template-id": "lt-07f8df694cf74912d"}
        2023-02-21T20:03:54.506Z	INFO	controller.provisioner.cloudprovider	launched new instance{"commit": "dedf6c1", "provisioner": "biterfossil-2-xzruescu8h", "id": "i-0e5c819e036d44099", "hostname": "ip-192-168-81-105.us-west-2.compute.internal", "instance-type": "g4dn.8xlarge", "zone": "us-west-2a", "capacity-type": "on-demand"}
        2023-02-21T20:04:58.128Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "7bcbad43-c526-49cc-b38b-346a7317fe6c", "error": "determining instance type, getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.134Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "429b7888-d8e4-49a7-a5b4-14a04ce37994", "error": "determining instance type, getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.145Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "94fdc58e-375d-4238-abbf-459faa51f83a", "error": "determining instance type, getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.148Z	INFO	controller.termination	cordoned node	{"commit": "dedf6c1", "node": "ip-192-168-81-105.us-west-2.compute.internal"}
        2023-02-21T20:04:58.149Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "eddc860e-7b16-4e09-8254-07351e174953", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.155Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "ed6d8c2d-2419-41b1-8e05-93a82e3e8d77", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.165Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "023b1d57-26e9-45d8-9625-f0c4ebf0e55a", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.166Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "26e04ffa-5c17-45e5-8aba-e83ed9ae33c1", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.179Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "8a001a72-1c60-4754-a55b-7a411cccd240", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.185Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "6c9380f1-6f8d-4a0b-8af6-07292d7cc85e", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.193Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "4cd42cdd-6a7e-4409-bde2-9c5d68d0ad99", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.346Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "5837db85-3c97-42f7-ace1-47ea23dc7196", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:04:58.987Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "08be331e-e583-4685-aa7a-e5038efe37f9", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:05:00.267Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "094e039f-c36a-462b-964c-0a50f38cf2bc", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:05:00.592Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "4bb2cea3-b15a-404e-95e4-8f5431f19885", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:05:00.606Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "16bec432-046f-4e63-9caf-0891243d5efe", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:05:02.828Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "75a67275-8ca2-45d7-869c-a5f8a170fbe4", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:05:05.434Z	ERROR	controller	Reconciler error	{"commit": "dedf6c1", "controller": "node-state", "controllerGroup": "", "controllerKind": "Node", "Node": {"name":"ip-192-168-81-105.us-west-2.compute.internal"}, "namespace": "", "name": "ip-192-168-81-105.us-west-2.compute.internal", "reconcileID": "1e8ab4d0-8fe0-430c-a2c6-152df63b87af", "error": "getting providerRef, AWSNodeTemplate.karpenter.k8s.aws \"stealerleaf-1-kgio9gihue\" not found"}
        2023-02-21T20:05:05.731Z	INFO	controller.termination	deleted node	{"commit": "dedf6c1", "node": "ip-192-168-81-105.us-west-2.compute.internal"}
------- END AFTER -------
• [77.880 seconds]
```

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [X] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by bwagner5 — COMMENTED** [review]
Nice! LGMT can you add to the docs table in concepts/scheduling too?

File: website/content/en/preview/concepts/scheduling.md [hunk 1]
```diff
@@ -148,6 +148,7 @@ Take care to ensure the label domains are correct. A well known label like `karp
 | karpenter.k8s.aws/instance-size                       | 8xlarge     | [AWS Specific] Instance types of similar resource quantities but different properties                                                       |
 | karpenter.k8s.aws/instance-cpu                        | 32          | [AWS Specific] Number of CPUs on the instance                                                                                               |
 | karpenter.k8s.aws/instance-memory                     | 131072      | [AWS Specific] Number of mebibytes of memory on the instance                                                                                |
+| karpenter.k8s.aws/instance-network                     | 131072      | [AWS Specific] Number of [baseline megabits](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-network-bandwidth.html) available on the instance                                                                                |
```
Comment (bwagner5) [inline website/content/en/preview/concepts/scheduling.md:151]:
```suggestion
| karpenter.k8s.aws/instance-network-bandwidth                     | 131072      | [AWS Specific] Number of [baseline megabits](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-network-bandwidth.html) available on the instance                                                                                |
```

**Review by bwagner5 — APPROVED** [review]
LGTM

**Comment by alex-berger** [comment]
I guess this fixed/implements https://github.com/aws/karpenter/issues/3238.

**Pull request merged**